src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -5159,6 +5185,33 @@ void Assembler::evpmadd52luq(XMMRegister dst, KRegister mask, XMMRegister src1,
   emit_int16((unsigned char)0xB4, (0xC0 | encode));
 }
 
+void Assembler::vpmadd52huq(XMMRegister dst, XMMRegister src1, Address src2, int vector_len) {
+  vpmadd52huq(dst, src1, src2, false, vector_len);
+}
+
+void Assembler::vpmadd52huq(XMMRegister dst, XMMRegister src1, Address src2, bool merge, int vector_len) {
+  assert(VM_Version::supports_avxifma() && (vector_len == AVX_128bit || vector_len == AVX_256bit), "");
+  InstructionMark im(this);
+  InstructionAttr attributes(vector_len, /* rex_w */ true, /* legacy_mode */ false, /* no_mask_reg */ false, /* uses_vl */ false);
```

Comment:
uses_vl should be true for all vpmadd52 instructions. 
The assert could be coded as:
assert(vector_len == AVX_512bit ? VM_Version::supports_evex() : VM_Version::supports_avxifma(), "");

Reply:
Please see the updated checks as suggested.